lib/jara/releaser.rb

```diff
@@ -165,12 +182,77 @@ def exec(command)
           raise ExecError, %(Command `#{command}` failed with output: #{output})
         end
         output
+      rescue Errno::ENOENT => e
+        raise ExecError, %(Command `#{command}` failed: #{e.message})
       end
     end
 
     class Archiver
+      def initialize(shell)
+        @shell = shell
+      end
+
+      def create(options)
+      end
+
+      def extension
+      end
+
+      def content_type
+      end
+
+      def metadata
+        {}
+      end
+    end
+
+    class Tarchiver < Archiver
       def create(options)
-        Puck::Jar.new(options).create!
+        FileUtils.mkdir_p('build')
+        entries = Dir['*']
+        entries.delete('build')
+        @shell.exec("tar czf build/#{options[:archive_name]} #{entries.join(' ')}")
+      end
+
+      def extension
+        'tgz'
+      end
+
+      def content_type
+        'application/x-gzip'
+      end
+    end
+
+    if defined? JRUBY_VERSION
+      begin
+        require 'puck'
+
+        class PuckArchiver < Archiver
+          def create(options)
+            options = options.dup
+            options[:jar_name] = options.delete(:archive_name)
+            Puck::Jar.new(options).create!
+          end
+
+          def extension
+            'jar'
+          end
+
+          def content_type
+            'application/java-archive'
+          end
+
+          def metadata
+            jruby_jars_path = $LOAD_PATH.grep(/\/jruby-jars/).first
+            jruby_version = jruby_jars_path && jruby_jars_path.scan(/\/jruby-jars-(.+)\//).flatten.first
+            if jruby_version
+              super.merge('jruby' => jruby_version)
+            else
+              super
+            end
+          end
+        end
+      rescue LoadError
```

Comment:
I'm not a fan of definitions within big begin/rescue-blocks. I would much rather see something along the lines of

```ruby
puck_enabled = false
if JRUBY_VERSION
  begin
    require 'puck'
    puck_enabled =  true
  rescue => e
    warn "Puck not available: #{e.message} (#{e.class})"
  end
end

if puck_enabled
  ...
end
```
I would also like to see some indication of what failed to load. I have debugged too many projects where a failure to load zlib or similar was hidden by the fact that (the wrong) `LoadError` was just silently captured.